actors/evm/src/interpreter/instructions/ext.rs

```diff
@@ -37,15 +46,36 @@ pub fn extcodehash(
     system: &System<impl Runtime>,
     addr: U256,
 ) -> Result<U256, StatusCode> {
-    let addr = match get_cid_type(system.rt, addr) {
-        ContractType::EVM(a) => Ok(a),
-        _ => Err(StatusCode::InvalidArgument(
-            "Cannot invoke EXTCODEHASH for non-EVM actor.".to_string(),
-        )),
-    }?;
-    let bytecode_cid = get_evm_bytecode_cid(system.rt, &addr)?;
+    let addr = match get_address_type(system.rt, addr) {
+        ContractType::EVM(a) => a,
+        // _Technically_ since we have native "bytecode" set as 0xfe this is valid, though we cant differentiate between
+        ContractType::Native(_) => return Ok(NATIVE_BYTECODE_HASH.into()),
+        // Precompiles "exist" and therefore aren't empty (although spec-wise they can be either 0 or keccak("") ).
+        ContractType::Precompile => return Ok(EMPTY_EVM_HASH.into()),
+        // Everything else is flattened to 0, "empty hash"
+        _ => return Ok(U256::zero()),
+    };
+
+    // TODO: We have no way telling if an account is "empty" as described in EIP-161 We can check for no code and zero balance,
+    //      but we cannot discern if the nonce is zero following semantics of EVM nonces. Therefore there may be accounts that in EVM would be
+    //      considered "empty" and give 0, but we will instead return keccak(""), indicating existing but empty bytecode
+    //      (tracking nonce be possible if we can run code on receive)
+    //
```

Comment:
please note this

Reply:
I think the answer here is "don't care". I.e., if it exists in state, it exists. If it doesn't exist in state, it doesn't exist.

Reply:
I.e., I'd just remove this TODO and note that we deviate from the EVM here by design.